File: statistics/TypeScript/gcd.ts
/**
 * gcd 最大公约数
 * @param  {number} p [description]
 * @param  {number} q [description]
 * @return {number}   [description]
 */
function gcd(p:number,q:number):number {
	if(q===0) return p;
	let r =p % q;
	return gcd(q,r);
}
export {gcd};

File: statistics/C/gcd.c
#include "gcd.h"
/**
 * 求最大公约数
 * @param  p [description]
 * @param  q [description]
 * @return   [description]
 */
int gcd(int p,int q){
    if(q==0) return p;
    int r = p % q;
    return gcd(q,r);
}